### ISSUE TITLE
Question about training tasks used.

### ISSUE THREAD

**Issue by zlm646** [InternRobotics/InternNav#51]
Hi:
Thanks for your excellent work. I justed noticed in NavDP training code that you only used no goal navigation task along with point goal navigation task, not like you mentioned in the paper that using all four tasks for training achieves the best result, can you explain the reason behind this mismatch please?

**Comment by wzcai99** [comment]
Hi, currently, the provided training code provides a minimal training validation, and without the other two losses, you can still get a satisfying policy performance. We will update the training code with the other two losses recently. Please stay tuned.

**Comment by zlm646** [comment]
> Hi, currently, the provided training code provides a minimal training validation, and without the other two losses, you can still get a satisfying policy performance. We will update the training code with the other two losses recently. Please stay tuned.

And I'm also wondering why the model generize well across different robot platforms since the robot size is not encoded into the model(what I mean is a narrow corridor may be traversable for Go2  but not for G1).

**Comment by wzcai99** [comment]
> > Hi, currently, the provided training code provides a minimal training validation, and without the other two losses, you can still get a satisfying policy performance. We will update the training code with the other two losses recently. Please stay tuned.
> 
> And I'm also wondering why the model generize well across different robot platforms since the robot size is not encoded into the model(what I mean is a narrow corridor may be traversable for Go2 but not for G1).

This is a good point. Currently, the cross-embodiment generalization is more like a cross-camera generalization. 
Some embodiment information can be reflected by the camera configuration, like the robot's height.

**Comment by wzcai99** [comment]
> Thanks for applying. By the way, I visualize the model inference result using the checkpoint you provided(navdp-weights.ckpt) with matterport3d_d435i dataset without point goal input, and here are a few samples which run into obstacles, is this reasonble? And which model do you use in the project page In-the-Wild Real-World Navigation demo video?
> 
> <img alt="Image" width="2000" height="2000" src="https://private-user-images.githubusercontent.com/175006920/485094241-ab65cfdf-9a3b-46c7-a626-ba1605e3704e.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTY5NTY2NDAsIm5iZiI6MTc1Njk1NjM0MCwicGF0aCI6Ii8xNzUwMDY5MjAvNDg1MDk0MjQxLWFiNjVjZmRmLTlhM2ItNDZjNy1hNjI2LWJhMTYwNWUzNzA0ZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwOTA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDkwNFQwMzI1NDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wZjk0ODI4YjY4YWI1MGVmZDhlNDc1MTMxYjUxYjFjOTMxZjM3YTg0MjU5MmZiODM1ZTE1MTNiYTBmNWQ5N2MyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.D3oooXgraXnEjlziHnap0u8to_qdN39UiFsQf65xzGQ"> <img alt="Image" width="2000" height="2000" src="https://private-user-images.githubusercontent.com/175006920/485094290-84b178ff-049a-419f-9f1c-25aff4503f2b.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTY5NTY2NDAsIm5iZiI6MTc1Njk1NjM0MCwicGF0aCI6Ii8xNzUwMDY5MjAvNDg1MDk0MjkwLTg0YjE3OGZmLTA0OWEtNDE5Zi05ZjFjLTI1YWZmNDUwM2YyYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwOTA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDkwNFQwMzI1NDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wMzI1ZjJlZDZjMGQ3ZDY1YzhiYmJiN2I3MWFiNjcwYWQyYmNlYzI0YWEzZTI0MTJmMTA3YzI1ZTk3ZjY3YzE1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.DnjR1rkDYHPSsR6laUIPX8OXMuL7V_jixrd0fnSDUhQ">

We use navdp-cross-modal.ckpt for the Real-World Navigation demo video.

**Comment by zlm646** [comment]
> > > Thanks for applying. By the way, I visualize the model inference result using the checkpoint you provided(navdp-weights.ckpt) with matterport3d_d435i dataset without point goal input, and here are a few samples which run into obstacles, is this reasonble? And which model do you use in the project page In-the-Wild Real-World Navigation demo video?
> > > <img alt="Image" width="2000" height="2000" src="https://private-user-images.githubusercontent.com/175006920/485094241-ab65cfdf-9a3b-46c7-a626-ba1605e3704e.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTY5NTY2NDAsIm5iZiI6MTc1Njk1NjM0MCwicGF0aCI6Ii8xNzUwMDY5MjAvNDg1MDk0MjQxLWFiNjVjZmRmLTlhM2ItNDZjNy1hNjI2LWJhMTYwNWUzNzA0ZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwOTA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDkwNFQwMzI1NDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wZjk0ODI4YjY4YWI1MGVmZDhlNDc1MTMxYjUxYjFjOTMxZjM3YTg0MjU5MmZiODM1ZTE1MTNiYTBmNWQ5N2MyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.D3oooXgraXnEjlziHnap0u8to_qdN39UiFsQf65xzGQ"> <img alt="Image" width="2000" height="2000" src="https://private-user-images.githubusercontent.com/175006920/485094290-84b178ff-049a-419f-9f1c-25aff4503f2b.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTY5NTY2NDAsIm5iZiI6MTc1Njk1NjM0MCwicGF0aCI6Ii8xNzUwMDY5MjAvNDg1MDk0MjkwLTg0YjE3OGZmLTA0OWEtNDE5Zi05ZjFjLTI1YWZmNDUwM2YyYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwOTA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDkwNFQwMzI1NDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wMzI1ZjJlZDZjMGQ3ZDY1YzhiYmJiN2I3MWFiNjcwYWQyYmNlYzI0YWEzZTI0MTJmMTA3YzI1ZTk3ZjY3YzE1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.DnjR1rkDYHPSsR6laUIPX8OXMuL7V_jixrd0fnSDUhQ">
> > 
> > 
> > We use navdp-cross-modal.ckpt for the Real-World Navigation demo video.
> 
> And the difference between these two model is?

Do you have a demo to inference with this model please?

**Comment by wzcai99** [comment]
They are trained with different datasets and the navdp-cross-modal is a better checkpoint.

**Comment by wzcai99** [comment]
Yes, we will release the training code with another two loss functions.

**Comment by wzcai99** [comment]
Yes, we will release the code about the other two loss functions before October.

**Comment by zlm646** [comment]
> They are trained with different datasets and the navdp-cross-modal is a better checkpoint.

@wzcai99 Could you please explain the difference bitween the two datasets used more in detail please?

**Comment by wzcai99** [comment]
The navdp-cross-modal is trained with navigation dataset generated within HM3D, MP3D, 3D-Front, Replica, Gibson, and HSSD scenes, but the previous model is trained with only a subset of them. In addition, we also add the goal encoding alignment loss for the navdp-cross-modal.ckpt, while the previous did not.
